sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/WindowsCredentialApi.java

```diff
@@ -0,0 +1,294 @@
+import com.sun.jna.win32.StdCallLibrary;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * This class exposes functions from credential manager on Windows platform
+ * via JNA.
+ */
+public interface WindowsCredentialApi extends StdCallLibrary {
+
+    /**
+     * Type of Credential
+     */
+    int CRED_TYPE_GENERIC = 1;
+
+    /**
+     * Credential attributes
+     *
+     * typedef struct _CREDENTIAL_ATTRIBUTE {
+     *   LPTSTR Keyword;
+     *   DWORD  Flags;
+     *   DWORD  ValueSize;
+     *   LPBYTE Value;
+     * } CREDENTIAL_ATTRIBUTE, *PCREDENTIAL_ATTRIBUTE;
+     *
+     */
+    class CREDENTIAL_ATTRIBUTE extends Structure {
+
+        public static class ByReference extends CREDENTIAL_ATTRIBUTE implements Structure.ByReference { }
+
+        @Override
+        protected List<String> getFieldOrder() {
+            return Arrays.asList("Keyword",
+                    "Flags",
+                    "ValueSize",
+                    "Value");
+        }
+
+        /**
+         *    Name of the application-specific attribute. Names should be of the form "CompanyName_Name".
+         *    This member cannot be longer than CRED_MAX_STRING_LENGTH (256) characters.
+         */
+        public String Keyword;
+
+        /**
+         *   Identifies characteristics of the credential attribute. This member is reserved and should be originally
+         *   initialized as zero and not otherwise altered to permit future enhancement.
+         */
+        public int Flags;
+
+        /**
+         *   Length of Value in bytes. This member cannot be larger than CRED_MAX_VALUE_SIZE (256).
+         */
+        public int ValueSize;
+
+        /**
+         *   Data associated with the attribute. By convention, if Value is a text string, then Value should not
+         *   include the trailing zero character and should be in UNICODE.
+         *
+         *   Credentials are expected to be portable. The application should take care to ensure that the data in
+         *   value is portable. It is the responsibility of the application to define the byte-endian and alignment
+         *   of the data in Value.
+         */
+        public Pointer Value;
+    }
+
+
+    /**
+     * The CREDENTIAL structure contains an individual credential
+     *
+     * typedef struct _CREDENTIAL {
+     *   DWORD                 Flags;
+     *   DWORD                 Type;
+     *   LPTSTR                TargetName;
+     *   LPTSTR                Comment;
+     *   FILETIME              LastWritten;
+     *   DWORD                 CredentialBlobSize;
+     *   LPBYTE                CredentialBlob;
+     *   DWORD                 Persist;
+     *   DWORD                 AttributeCount;
+     *   PCREDENTIAL_ATTRIBUTE Attributes;
+     *   LPTSTR                TargetAlias;
+     *   LPTSTR                UserName;
+     * } CREDENTIAL, *PCREDENTIAL;
+     */
+    class CREDENTIAL extends Structure {
```

Comment:
Same here - class names should use Pascal case.

Reply:
Maps to native C structs.